app/lib/actions/middleware/backend_services_check.rb

```diff
@@ -0,0 +1,39 @@
+module Actions
+  module Middleware
+    class BackendServicesCheck < Dynflow::Middleware
+      def plan(*args)
+        #To prevent the ping from happening multiple times, keep track on the initial entry action
+        parent = source_action
+        parent.input[:services_checked] ||= []
+        to_check = services - parent.input[:services_checked]
+
+        if Setting[:check_services_before_actions] && to_check.any?
```

Comment:
Last nitpick, the setting check should probably be at the very top of htis method, so when turned off, we don't touch the input nor we search for `source_action`